internal/core/commands/sync_dd_power_train_type.go

```diff
@@ -0,0 +1,201 @@
+package commands
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"os"
+
+	coremodels "github.com/DIMO-Network/device-definitions-api/internal/core/models"
+	"github.com/DIMO-Network/device-definitions-api/internal/infrastructure/db/models"
+	"github.com/DIMO-Network/shared/db"
+	"github.com/TheFellow/go-mediator/mediator"
+	"github.com/rs/zerolog"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+	"gopkg.in/yaml.v3"
+)
+
+type SyncPowerTrainTypeCommand struct {
+}
+
+type SyncPowerTrainTypeCommandResult struct {
+	Status bool
+}
+
+func (*SyncPowerTrainTypeCommand) Key() string { return "SyncPowerTrainTypeCommand" }
+
+type SyncPowerTrainTypeCommandHandler struct {
+	DBS    func() *db.ReaderWriter
+	logger zerolog.Logger
+}
+
+func NewSyncPowerTrainTypeCommandHandler(dbs func() *db.ReaderWriter, logger zerolog.Logger) SyncPowerTrainTypeCommandHandler {
+	return SyncPowerTrainTypeCommandHandler{DBS: dbs, logger: logger}
+}
+
+func (ch SyncPowerTrainTypeCommandHandler) Handle(ctx context.Context, _ mediator.Message) (interface{}, error) {
+
+	all, err := models.DeviceDefinitions(models.DeviceDefinitionWhere.Verified.EQ(true),
+		qm.Load(models.DeviceDefinitionRels.DeviceStyles),
+		qm.Load(models.DeviceDefinitionRels.DeviceType),
+		qm.Load(models.DeviceDefinitionRels.DeviceMake)).All(ctx, ch.DBS().Reader)
```

Comment:
add a where the device_type is vehicle etc. Maybe query the device_types table and find the one that has powertrain_type as an attribute and pick that key?